### PR TITLE
chore: Clean up makefile

### PR DESCRIPTION
**Pull request by emgeee**
The current guide for setting up a development environment is unclear and the existing makefile includes multiple tasks that do very similar things, are un-used, or are confusingly named. This PR seeks to clear up confusion around how one should configure their python environment for developing feast.

This PR
- cleans up makefile tasks related to python dependencies to make it more clean which should be used when
- Updates the environment setup docs to make it clear UV should be used for dependency management

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,50 +41,50 @@ protos: compile-protos-python compile-protos-docs
 
 build: protos build-java build-docker
 
-# Python SDK
+# Python SDK - local
+# formerly install-python-ci-dependencies-uv-venv
+# editable install
+install-python-dependencies-dev:
```
Comment (franciscojavierarceo) [inline Makefile:47]: nice

File: Makefile [hunk 2]
```diff
@@ -35,50 +41,50 @@ protos: compile-protos-python compile-protos-docs
...
-	uv pip install --system --no-deps -e .
-
-install-python-ci-dependencies-uv-venv:
-	uv pip sync sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
-	uv pip install --no-deps -e .
-
-lock-python-ci-dependencies:
-	uv pip compile --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py$(PYTHON_VERSION)-ci-requirements.txt
-
-compile-protos-python:
```
Comment (franciscojavierarceo) [inline Makefile:63]: did you mean to remove this one?

File: Makefile [hunk 3]
```diff
@@ -35,50 +41,50 @@ protos: compile-protos-python compile-protos-docs
...
-	pixi run --environment py310 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile -p 3.10 --system --no-strip-extras setup.py --output-file sdk/python/requirements/py3.10-requirements.txt"
-	pixi run --environment py310 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile -p 3.10 --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py3.10-ci-requirements.txt"
-	pixi run --environment py311 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile -p 3.11 --system --no-strip-extras setup.py --output-file sdk/python/requirements/py3.11-requirements.txt"
-	pixi run --environment py311 --manifest-path infra/scripts/pixi/pixi.toml "uv pip compile -p 3.11 --system --no-strip-extras setup.py --extra ci --output-file sdk/python/requirements/py3.11-ci-requirements.txt"
+	# Remove all existing requirements because we noticed the lock file is not always updated correctly.
+	# Removing and running the command again ensures that the lock file is always up to date.
+	rm -rf sdk/python/requirements/* 2>/dev/null || true
+
+	$(foreach ver,$(PYTHON_VERSIONS),\
```
Comment (franciscojavierarceo) [inline Makefile:76]: nice

File: Makefile [hunk 4]
```diff
@@ -35,50 +41,50 @@ protos: compile-protos-python compile-protos-docs
...
+		pixi run --environment $(call get_env_name,$(ver)) --manifest-path infra/scripts/pixi/pixi.toml \
+			"uv pip compile -p $(ver) --system --no-strip-extras setup.py \
+			--output-file sdk/python/requirements/py$(ver)-requirements.txt" && \
+		pixi run --environment $(call get_env_name,$(ver)) --manifest-path infra/scripts/pixi/pixi.toml \
+			"uv pip compile -p $(ver) --system --no-strip-extras setup.py --extra ci \
+			--output-file sdk/python/requirements/py$(ver)-ci-requirements.txt" && \
+	) true
+
+
+compile-protos-python:
```
Comment (franciscojavierarceo) [inline Makefile:86]: oh i see you moved them here

Comment (emgeee) [reply Makefile:86]: yeah, just trying to organize it a little better

File: docs/project/development-guide.md [hunk 5]
```diff
@@ -120,51 +120,39 @@ Note that this means if you are midway through working through a PR and rebase,
...
-  ```
-4. (M1 Mac only): Follow the [dev guide](https://github.com/feast-dev/feast/issues/2105)
-5. Install uv. It is recommended to use uv for managing python dependencies.
+- `make` is used to run various scripts
+- [uv](https://docs.astral.sh/) for managing python dependencies. [installation instructions](https://docs.astral.sh/uv/getting-started/installation/)
+- (M1 Mac only): Follow the [dev guide if you have issues](https://github.com/feast-dev/feast/issues/2105)
+- (Optional): Node & Yarn (needed for building the feast UI)
+- (Optional): [Pixi](https://pixi.sh/latest/) for recompile python lock files. Only when you make changes to requirements or simply want to update python lock files to reflect latest versioons.
+
+### Quick start
```
Comment (franciscojavierarceo) [inline docs/project/development-guide.md:133]: nice

**Review by franciscojavierarceo — APPROVED** [review]
this lgtm

FYI @HaoXuAI @tokoko @shuchu @dmartinol @lokeshrangineni

**Comment by franciscojavierarceo** [comment]
Looks like pytest failed because it's not installed in the dependencies. May have to add that to the action.

**Review by dmartinol — APPROVED** [review]

**Pull request merged**